src/custom/utils/tip.test.ts

```diff
@@ -0,0 +1,268 @@
+  const feeAsCurrency = tryParseAmount(minimalFee, inputCurrency)
+
+  // Type checks mainly
+  if (!typedAmountAsCurrency || !pair || !feeAsCurrency) return null
+
+  // Calculate IN // OUT trades
+  let trade: Trade & {
+    inputAmountWithFees?: CurrencyAmount
+  }
+  if (isExactIn) {
+    // subtract fee from typed in amount
+    const typedAmountWithFees = typedAmountAsCurrency.subtract(feeAsCurrency)
+    const preTrade =
+      // This is how Uni calculates trades in their hook and in Trades.ts
+      Trade.bestTradeExactIn(
+        [pair1and2],
+        // subtract fee from typed in amount
+        typedAmountWithFees,
+        outputCurrency,
+        DEFAULT_HOPS
+        // Yes, this is also copy/paste
+      )[0] ?? null
+
+    // We need to iverride the Trade object to use different values as we are intercepting initial inputs
+    // and applying fee. For ExactIn orders, we leave outputAmount as is
+    // and only change inputAmount to show the original entry before fee calculation
+    trade = {
+      ...preTrade,
+      get inputAmount() {
+        return typedAmountAsCurrency!
+      },
+      get minimumAmountOut() {
+        return this.minimumAmountOut
+      },
+      get maximumAmountIn() {
+        return this.maximumAmountIn
+      },
+      inputAmountWithFees: typedAmountWithFees
+    }
+  } else {
+    // This is how Uni calculates trades in their hook and in Trades.ts
+    // Yes, the ?? null is also copy/paste
+    const preTrade = Trade.bestTradeExactOut([pair1and2], inputCurrency, typedAmountAsCurrency, DEFAULT_HOPS)[0] ?? null
+
+    const inputAmountWithFee = preTrade.inputAmount!.add(feeAsCurrency!)
+    // We need to iverride the Trade object to use different values as we are intercepting initial inputs
+    // and applying fee. For ExactOut orders, we leave inputAmount as is
+    // and only change outputAm to show the original entry before fee calculation
+    trade = {
+      ...preTrade,
+      inputAmount: inputAmountWithFee,
+      get minimumAmountOut() {
+        return this.minimumAmountOut
+      },
+      get maximumAmountIn() {
+        return this.maximumAmountIn
+      }
+    }
+  }
+
+  return {
+    ...props,
+    trade,
+    typedAmountAsCurrency,
+    feeAsCurrency
+  }
+}
+
+// CONFIG
+const typedAmount = (Math.random() * 9382).toString()
+const minimalFee = (Math.random() * 100).toString()
+
+describe('prices', () => {
+  describe('computeTradePriceBreakdown', () => {
+    // Temp test variables
+    let exactInOutputAmount: CurrencyAmount
+
+    it(`EXACT-IN TRADE: ${minimalFee} minimalFee - selling ${typedAmount}`, () => {
+      const tradeData = setupTrade({
+        type: TradeType.EXACT_INPUT,
+        typedAmount,
+        fee: {
+          minimalFee
+        },
+        inputCurrency: token1,
+        outputCurrency: token2,
+        pair: pair1and2
+      })
+
+      if (!tradeData?.trade) throw new Error('Test failed::Trade returned NULL')
```

Comment:
Do you expect the trade to fail?
If so, create a test for it, instead of a conditional inside a test.

Reply:
i don't, but just wanted a log for testing. good point tho